csrc/multidevice/communicator.h

```diff
@@ -6,15 +6,51 @@
  */
 // clang-format on
 #pragma once
-#ifdef NVFUSER_DISTRIBUTED
+
+#include <ATen/core/TensorBody.h>
+#include <ATen/core/ivalue.h>
+#include <c10/util/intrusive_ptr.h>
 
 #include <exceptions.h>
 #include <multidevice/multidevice.h>
-#include <torch/csrc/distributed/c10d/ProcessGroup.hpp>
-#include <torch/csrc/distributed/c10d/Store.hpp>
+#ifdef NVFUSER_DISTRIBUTED
+#include <torch/csrc/distributed/c10d/Backend.hpp>
 #include <torch/csrc/distributed/c10d/TCPStore.hpp>
+#include <torch/csrc/distributed/c10d/Work.hpp>
+#endif
 #include <visibility.h>
 
+#if !defined(NVFUSER_DISTRIBUTED)
+namespace c10d {
+class Work : public torch::CustomClassHolder {
+ public:
+  void wait() {}
+};
+
+class Backend : public torch::CustomClassHolder {
+ public:
+  c10::intrusive_ptr<c10d::Work> barrier() {
+    return c10::make_intrusive<c10d::Work>();
+  }
+  c10::intrusive_ptr<c10d::Work> send(
+      std::vector<at::Tensor>& tensors,
+      int dstRank,
+      int tag) {
+    return c10::make_intrusive<c10d::Work>();
+  }
+  c10::intrusive_ptr<Work> recv(
```

Comment:
```suggestion
  c10::intrusive_ptr<c10d::Work> recv(
```

nit: just to stay consistent with the other return types

Reply:
Thanks! I instead made others `Work` because it's under the c10d namespace. I also moved the mocks to a separate header file.